hack/shellcheck.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+if [ "$IS_CONTAINER" != "" ]; then
+  find ./ ! -name "$(printf "*\\n*")" -name '*.sh' > tmp-search
```

Comment:
Do we need the newline guard?  I don't expect us to have any files ending in `.sh` with shell-sensitive characters in the names.  I expect we can also drop the `tmp-search` business and just paste in the:

```sh
for file in $(find /workdir/ -type f -name "*.sh"); do
```

from `.travis.yaml`.

Reply:
In fact, we can probably just do:

```sh
shellcheck --format=gcc $(find "${DIR}" -type f -name '*.sh')
```

and skip the `for` loop entirely (although this diverges from `.travis.yaml`, so I'm fine punting it to follow-up work).

If we replace `/workdir/` with `"${DIR}"` in the `find` (to allow folks to call this locally without spinning up a container), you'd want to adjust the container call to:

```sh
docker run -e IS_CONTAINER='TRUE' --rm -v "$(realpath "${DIR}")":/workdir:ro --entrypoint sh quay.io/coreos/shellcheck-alpine:v0.5.0 /workdir/hack/shellcheck.sh /workdir;
```

and add a leading:

```sh
DIR="${1:-.}"
```

or similar to the script ([`realpath` is not in POSIX][1], but it is in coreutils).

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/idx/utilities.html